File: android/http-core/src/main/java/com/caldremch/http/core/params/HttpParams.java
package com.caldremch.http.core.params;

import org.jetbrains.annotations.NotNull;

import java.io.File;
import java.util.LinkedHashMap;
import java.util.Map;

/**
 * Created by Leon on 2022/10/30.
 */
public class HttpParams {

    @NotNull
    private Map<String, Object> urlParamsMap = new LinkedHashMap();

    @NotNull
    public final Map<String, Object> getUrlParamsMap() {
        return this.urlParamsMap;
    }

    @NotNull
    public final Map<String, Object> getUrlParams() {
        return this.urlParamsMap;
    }

    public final void setUrlParamsMap(@NotNull Map<String, Object> map) {
        this.urlParamsMap = map;
    }

    public HttpParams() {
    }

    public HttpParams(@NotNull String key, @NotNull String value) {
        put(key, value);
    }

    public HttpParams(@NotNull String key, @NotNull File file) {
        put(key, file);
    }

    public final void put(@NotNull String key, @NotNull Object value) {
        this.urlParamsMap.put(key, value);
    }

    public final boolean isEmpty() {
        return this.urlParamsMap.isEmpty();
    }
}

File: http-core/src/main/java/com/caldremch/http/core/params/HttpParams.kt
package com.caldremch.http.core.params

import java.io.File

/**
 * @author Caldremch
 * @date 2019-08-20 15:54
 * @email caldremch@163.com
 * @describe
 */
class HttpParams {
   private var urlParamsMap: MutableMap<String, Any> = mutableMapOf()

    fun getUrlParamsMap():MutableMap<String, Any> {
       return urlParamsMap
    }

    val urlParams: MutableMap<String, Any>
        get() = urlParamsMap

    fun setUrlParamsMap(urlParamsMap: MutableMap<String, Any>) {
        this.urlParamsMap = urlParamsMap
    }

    constructor() {
    }

    constructor(key: String, value: String) {
        put(key, value)
    }

    constructor(key: String, file: File) {
        put(key, file)
    }


    fun put(key: String, value: Any) {
        urlParamsMap[key] = value
    }

    val isEmpty: Boolean
        get() = urlParamsMap.isEmpty()


}